src/main/java/com/google/devtools/build/lib/packages/StructImpl.java

```diff
@@ -333,10 +327,16 @@ private void printJson(Object value, StringBuilder sb, Location loc, String cont
     }
   }
 
-  private String jsonEscapeString(String string) {
-    return escapeDoubleQuotesAndBackslashesAndNewlines(string)
+  private void jsonEscapeString(StringBuilder stringBuilder, String string, boolean appendColon) {
+    stringBuilder.append("\"");
```

Comment:
If you like, use append('"') to avoid the internal loop.

Reply:
Not sure I completely understand your suggestion. Was my change what you had in mind?